### PR TITLE
add new testing tool APKLab

### LINKED ISSUE #2158: [Tool] APKLab
Evaluate for inclusion:

https://github.com/Surendrajat/APKLab/

### PR DESCRIPTION
**Pull request by fujiokayu**
Thank you for submitting a Pull Request to the Mobile Security Testing Guide. Please make sure that:

- [x] Your contribution is written in the 2nd person (e.g. you)
- [x] Your contribution is written in an active present form for as much as possible.
- [x] You have made sure that the reference section is up to date (e.g. please add sources you have used, make sure that the references to MITRE/MASVS/etc. are up to date)
- [x] Your contribution has proper formatted markdown and/or code
- [x] Any references to website have been formatted as [TEXT](URL “NAME”)
- [x] You verified/tested the effectiveness of your contribution (e.g.: is the code really an effective remediation? Please verify it works!)

If your PR is related to an issue. Please end your PR test with the following line:
This PR closes #2158 .

Feel free to correct my English when it is wrong.

---

The Tool APKLab also provides an automated solution for the [Patching, Repackaging, and Re-Signing](https://github.com/OWASP/owasp-mstg/blob/master/Document/0x05c-Reverse-Engineering-and-Tampering.md#patching-repackaging-and-re-signing) section.

However, I have not added it here for the following reasons: 
- VScode is required as the execution environment.
- This section should provide more primitive knowledge.

### CODE REVIEW

File: Document/0x05c-Reverse-Engineering-and-Tampering.md [hunk 1]
```diff
@@ -34,6 +34,7 @@ If you want to look directly into Java source code on a GUI, simply open your AP
 
 Android decompilers go one step further and attempt to convert Android bytecode back into Java source code, making it more human-readable. Fortunately, Java decompilers generally handle Android bytecode well. The above mentioned tools embed, and sometimes even combine, popular free decompilers such as:
 
+- [APKLab](https://github.com/APKLab/APKLab "APKLab")
```
Comment (cpholguera) [inline Document/0x05c-Reverse-Engineering-and-Tampering.md:37]: Could you please remove this line and include APKLab below as:

Alternatively you can use the [APKLab](0x08a-Testing-Tools.md#apklab) extension for Visual Studio Code or run [apkx](0x08a-Testing-Tools.md#apkx) on your APK or use the exported files from the previous tools to open the reversed source code on your preferred IDE.

File: Document/0x08a-Testing-Tools.md [hunk 2]
```diff
@@ -1109,6 +1109,16 @@ For more information on what this tool can be used for, check out:
...
 
+### APKLab
+
+[APKLab](https://github.com/APKLab/APKLab "APKLab") is a Visual Studio Code extension that can unpack APKs, decompile DEX, apply patch, and rebuild APKs.  
+Internally it uses [apktool](https://github.com/iBotPeaches/Apktool "apktool") and [jadx](https://github.com/skylot/jadx "jadx"), so you can use those features in Visual Studio Code.
+
+In addition, APKLab provides various features such as signing APKs, MITM Patch, and more.  
+For more information, you can refer to the following documentation.
+
+- APKLab - <https://apklab.surendrajat.xyz/>
```
Comment (cpholguera) [inline Document/0x08a-Testing-Tools.md:1120]:
```suggestion
For more information, you can refer to [APKLab's official documentation](https://apklab.surendrajat.xyz/).
```

File: Document/0x08a-Testing-Tools.md [hunk 3]
```diff
@@ -1109,6 +1109,16 @@ For more information on what this tool can be used for, check out:
 - [APKiD: PEiD for Android Apps](https://github.com/enovella/cve-bio-enovella/blob/master/slides/bheu18-enovella-APKID.pdf)
 - [APKiD: Fast Identification of AppShielding Products](https://github.com/enovella/cve-bio-enovella/blob/master/slides/APKiD-NowSecure-Connect19-enovella.pdf)
 
+### APKLab
+
+[APKLab](https://github.com/APKLab/APKLab "APKLab") is a Visual Studio Code extension that can unpack APKs, decompile DEX, apply patch, and rebuild APKs.  
+Internally it uses [apktool](https://github.com/iBotPeaches/Apktool "apktool") and [jadx](https://github.com/skylot/jadx "jadx"), so you can use those features in Visual Studio Code.
```
Comment (cpholguera) [inline Document/0x08a-Testing-Tools.md:1115]:
```suggestion
[APKLab](https://github.com/APKLab/APKLab "APKLab") is a convenient Visual Studio Code extension leveraging tools such as [apktool](#apktool) and [jadx](#jadx) to enable features including app unpacking, decompilation, code patching (e.g. for MITM), and repackaging straight from the IDE.
```

File: Document/0x08a-Testing-Tools.md [hunk 4]
```diff
@@ -1109,6 +1109,16 @@ For more information on what this tool can be used for, check out:
...
+
+In addition, APKLab provides various features such as signing APKs, MITM Patch, and more.
```
Comment (cpholguera) [inline Document/0x08a-Testing-Tools.md:1117]:
```suggestion
```

**Review by cpholguera — CHANGES_REQUESTED** [review]
@fujiokayu I've included some suggestions to compact the text a little bit. Please feel free to apply them by clicking on "Commit suggestion" or start a discussion in the comments. Thank you very much for taking the time to test the tool and writing the PR!

File: Document/0x05c-Reverse-Engineering-and-Tampering.md (hunk 1 above)
Comment (fujiokayu) [reply Document/0x05c-Reverse-Engineering-and-Tampering.md]: fixed: 86fd672

**Comment by fujiokayu** [comment]
Thanks @cpholguera for checking this PR.
I agree with all your suggestions. Fixed them.

File: Document/0x05c-Reverse-Engineering-and-Tampering.md [hunk 5]
```diff
@@ -40,7 +40,7 @@ Android decompilers go one step further and attempt to convert Android bytecode
 - [Procyon](https://github.com/mstrobel/procyon "Procyon")
 - [CFR](https://www.benf.org/other/cfr/ "CFR")
 
-Alternatively run [apkx](0x08a-Testing-Tools.md#apkx) on your APK or use the exported files from the previous tools to open the Java source code in another tool such as an IDE.
+Alternatively you can use the [APKLab](https://github.com/APKLab/APKLab) extension for Visual Studio Code or run [apkx](0x08a-Testing-Tools.md#apkx) on your APK or use the exported files from the previous tools to open the reversed source code on your preferred IDE.
```
Comment (cpholguera) [inline Document/0x05c-Reverse-Engineering-and-Tampering.md:43]:
```suggestion
Alternatively you can use the [APKLab](0x08a-Testing-Tools.md#apklab) extension for Visual Studio Code or run [apkx](0x08a-Testing-Tools.md#apkx) on your APK or use the exported files from the previous tools to open the reversed source code on your preferred IDE.
```

**Review by cpholguera — APPROVED** [review]
Thanks a lot @fujiokayu!

**Pull request merged**